pkg/httpsproxy/server/model.go

```diff
@@ -27,3 +38,53 @@ type HttpApiResponseMessage struct {
 	RequestID      string
 	Headers        map[string]string
 }
+
+func NewHttpApiReqMessageFromNatsMessage(m *nats.Msg) (*HttpApiReqMessage, error) {
+	req := &HttpApiReqMessage{}
+	if err := json.Unmarshal(m.Data, &req); err != nil {
+		return nil, err
+	}
+	return req, nil
+}
+
+func (req *HttpApiReqMessage) ToHttpRequest() *http.Request {
+	url := fmt.Sprintf("http://%s%s", req.Target, req.HttpPath)
+	reader := bytes.NewReader(req.Body)
+	httpRequest, _ := http.NewRequest(req.HttpMethod, url, reader)
+	for _, x := range req.Headers {
+		httpRequest.Header.Add(x.Key, x.Values[0])
+	}
+
+	return httpRequest
+}
+
+func NewHttpApiResponseMessageFromHttpResponse(resp *http.Response) *HttpApiResponseMessage {
+	output := &HttpApiResponseMessage{}
+
+	output.HttpStatusCode = resp.StatusCode
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err == nil {
+		output.RespBody = string(bodyBytes)
+	} else {
+		log.WithError(err).Errorf("Got an error reading a resp body from http api")
+		output.RespBody = err.Error()
+	}
+	output.Headers = make(map[string]string)
+
+	for k, v := range resp.Header {
+		if len(v) > 0 {
+			output.Headers[k] = v[0]
+		} else {
+			output.Headers[k] = ""
+		}
+	}
```

Comment:
is looping over `resp.Header` necessary? Doesn't look like we actually change anything. Can't we just do `output.Headers = resp.Header`?

Reply:
definition of Header from the official pkg: `type Header map[string][]string`
vs Header from HttpApiResponseMessage: `Headers map[string]string`
Not sure if there is a better way to do this without iterating.

Reply:
Ok, yeah. I see now, we are indexing into v.